File: back-end/models/Payment.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Payment struct {
	ID uint `gorm:"primaryKey" json:"id"`

	// user id stored as uuid string to match User.ID
	UserID string `gorm:"type:uuid;not null;index" json:"user_id"`

	Name  string `gorm:"type:varchar(100);not null" json:"name"`
	Price int    `gorm:"not null" json:"price"`

	Situation string `gorm:"type:varchar(50);not null" json:"situation"`

	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

File: back-end/fluxopay-api/src/main/java/com/fintech/models/Payment.java
package com.fintech.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.Where;

import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@SQLDelete(sql = "UPDATE payments SET deleted_at = NOW() WHERE id = ?")
@Where(clause = "deleted_at IS NULL")

public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = false)
    private String userId;

    @Column(name = "customer_id")
    private Long customerId;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false)
    private Integer price;

    @Column(length = 100)
    private String description;

    @Column(nullable = false, length = 50)
    private String situation;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updateAt;

    private LocalDateTime deletedAt;
}
